devicetypes/qubino/qubino-flush-dimmer.src/qubino-flush-dimmer.groovy

```diff
@@ -0,0 +1,890 @@
+	def key = preference.key
+	def preferenceValue = getPreferenceValue(preference, cmd.scaledConfigurationValue)
+	if (settings."$key" == preferenceValue) {
+		state.currentPreferencesState."$key".value = settings."$key"
+		state.currentPreferencesState."$key".status = "synced"
+	} else if (preference.type == "boolRange") {
+		if (state.currentPreferencesState."$key".status == "disablePending" && preferenceValue == preference.disableValue) {
+			state.currentPreferencesState."$key".status = "disabled"
+		} else {
+			runIn(5, "syncConfiguration", [overwrite: true])
+		}
+	} else {
+		state.currentPreferencesState."$key"?.status = "syncPending"
+		runIn(5, "syncConfiguration", [overwrite: true])
+	}
+	// Preferences template end
+}
+
+private getPreferenceValue(preference, value = "default") {
+	def integerValue = value == "default" ? preference.defaultValue : value.intValue()
+	switch (preference.type) {
+		case "enum":
+			return String.valueOf(integerValue)
+		case "boolean":
+			return String.valueOf(preference.optionActive == integerValue)
+		default:
+			return integerValue
+	}
+}
+
+private getCommandValue(preference) {
+	def parameterKey = preference.key
+	switch (preference.type) {
+		case "boolean":
+			return settings."$parameterKey" ? preference.optionActive : preference.optionInactive
+		case "boolRange":
+			def parameterKeyBoolean = parameterKey + "Boolean"
+			return !notNullCheck(settings."$parameterKeyBoolean") || settings."$parameterKeyBoolean" ? settings."$parameterKey" : preference.disableValue
+		case "range":
+			return settings."$parameterKey"
+		default:
+			return Integer.parseInt(settings."$parameterKey")
+	}
+}
+
+private notNullCheck(value) {
+	return value != null
+}
+
+private isPreferenceChanged(preference) {
+	if (notNullCheck(settings."$preference.key")) {
+		if (preference.type == "boolRange") {
+			def boolName = preference.key + "Boolean"
+			if (state.currentPreferencesState."$preference.key".status == "disabled") {
+				return settings."$boolName"
+			} else {
+				return state.currentPreferencesState."$preference.key".value != settings."$preference.key" || !settings."$boolName"
+			}
+		} else {
+			return state.currentPreferencesState."$preference.key".value != settings."$preference.key"
+		}
+	} else {
+		return false
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd, ep = null) {
+	// Handles all Z-Wave commands we aren't interested in
+	log.debug("zwaveEvent other commands: ${cmd}")
+	[:]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd, ep = null) {
+	log.debug "BasicReport: ${cmd}"
+	dimmerEvents(cmd)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicSet cmd, ep = null) {
+	log.debug "BasicSet: ${cmd}"
+	dimmerEvents(cmd)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelReport cmd, ep = null) {
+	/*
+	This cmd value reports inaccurate values which often differ from the values reported by SwitchMultilevelSet by 1-3%.

```

Comment:
Are all those SwitchMultilevelReports differ 1-3 from SwitchMultilevelSet by 1-3% ?
(In a full scope, when You set level 0-100) or is it just a matter of certain narrow ranges ?

Reply:
@KKlimczukS Yes. While checking, all those SwitchMultilevelReports differ from SwitchMultilevelSet and that problem occurs in a full scope (0-100)